src/renderer/components/inputs/NumberInput.tsx

```diff
@@ -46,36 +52,81 @@ const NumericalInput = memo(
             setInputString(numberAsString);
 
             if (!Number.isNaN(numberAsNumber)) {
-                if (type.includes('odd')) {
-                    // Make the number odd if need be
-                    // round up the nearest odd number
-                    // eslint-disable-next-line no-param-reassign
-                    numberAsNumber += 1 - (numberAsNumber % 2);
-                }
                 setInput(numberAsNumber);
             }
         };
 
+        const onBlur = () => {
+            const valAsNumber =
+                precision > 0 ? parseFloat(inputString) : Math.round(parseFloat(inputString));
+
+            if (!Number.isNaN(valAsNumber)) {
+                const roundedVal = +`${Math.round(
+                    +`${Math.round((valAsNumber - offset) / step) * step + offset}e${precision}`
+                )}e-${precision}`;
```

Comment:
Why don't we just use the same formula as we use on the python side instead of this abomination?
```suggestion
                const roundedVal = Math.round(
                    (Math.round((valAsNumber - offset) / step) * step + offset) * 10 ** precision
                ) * 10 ** -precision;
```

Reply:
lol